bundles/org.openhab.binding.speedtest/src/main/java/org/openhab/binding/speedtest/internal/SpeedtestHandler.java

```diff
@@ -331,8 +349,19 @@ private void getSpeed() {
      */
     private void updateChannels() {
         logger.debug("Updating channels");
+        State newState;
 
-        State newState = new QuantityType<>(Double.parseDouble(pingJitter) / 1000.0, Units.SECOND);
+        // timestamp format: "2023-07-20T19:34:54Z"
+        try {
+            ZonedDateTime zonedDateTime = ZonedDateTime.parse(timestamp).withZoneSameInstant(ZoneId.systemDefault());
```

Comment:
If the timestamp format ends on "Z", i'd expect to be UTC. At least common Standards work that way. Also, i think parsing the timestamp (and other fields) should not be part of the channel update, but instead part of the validation of your ResultContainer. That would make your parsing testable and you can easier react of parsing failures without modifying the handler itself.

If you need TZ awareness, OpenHAB provides a `TimeZoneProvider` as service.

Reply:
> If you need TZ awareness, OpenHAB provides a TimeZoneProvider as service.

OK.
Yes, the string is UTC time. I'm using now `TimeZoneProvider` instead of `systemDefault()` for the transformation to local time.

> but instead part of the validation of your ResultContainer

Right now, the `ResultContainer` is just doing the "json 2 string" parameter transformation. All parameters from `ResultContainer` are provided as  `String` - no channel logic is implemented there. I've just inherited the code from the inital developer and do not intend to fully rewrite it.
I moved now all the data logic into `getSpeed()` and `updateChannels()` is now justs updating the channels w/o any logic.